components/automate-cli/cmd/chef-automate/certRotate.go

```diff
@@ -483,21 +523,48 @@ func patchOSNodeDN(flagsObj *certRotateFlags, patchFnParam *patchFnParameters, c
 	return nil
 }
 
-func getOldCn(automatesConfig map[string]*deployment.AutomateConfig) string {
+func (c *certRotateFlow) getFrontIpsToSkipRootCAandCNPatchingForOs(automatesConfig map[string]*deployment.AutomateConfig, newRootCA string, newCn string, node string, infra *AutomateHAInfraDetails) []string {
+	oldRootCA := ""
 	oldCn := ""
-	for _, config := range automatesConfig {
+	skipIpsList := []string{}
+	for ip, config := range automatesConfig {
 		if config.Global.V1.External.Opensearch != nil && config.Global.V1.External.Opensearch.Ssl != nil {
 			if config.Global.V1.External.Opensearch.Ssl.ServerName != nil {
 				oldCn = config.Global.V1.External.Opensearch.Ssl.ServerName.Value
-				break
+				if c.getFrontEndIpsForSkippingCnAndRootCaPatching(newRootCA, newCn, oldCn, oldRootCA, node) {
+					skipIpsList = append(skipIpsList, ip)
+				}
+			}
+			if config.Global.V1.External.Opensearch.Ssl.RootCert != nil {
+				oldRootCA = config.Global.V1.External.Opensearch.Ssl.RootCert.Value
+				if c.getFrontEndIpsForSkippingCnAndRootCaPatching(newRootCA, newCn, oldCn, oldRootCA, node) {
+					skipIpsList = append(skipIpsList, ip)
+				}
 			}
 		}
 	}
-	return oldCn
+	return skipIpsList
+}
+
+// getFrontIpsToSkipRootCAPatchingForPg compare new root-ca and current root-ca of remoteService and returns ips to skip root-ca patching.
+func (c *certRotateFlow) getFrontIpsToSkipRootCAPatchingForPg(automatesConfig map[string]*deployment.AutomateConfig, newRootCA string, infra *AutomateHAInfraDetails) []string {
+	oldRootCA := ""
+	skipIpsList := []string{}
+	for ip, config := range automatesConfig {
+		if config.Global.V1.External.Postgresql != nil && config.Global.V1.External.Postgresql.Ssl != nil {
+			if config.Global.V1.External.Postgresql.Ssl.RootCert != nil {
```

Comment:
You can avoid this nesting of inner if `config.Global.V1.External.Postgresql.Ssl != nil` outside

Reply:
done